kobo/hub/models.py

```diff
@@ -544,6 +550,11 @@ def __unicode__(self):
             return u"#%s [method: %s, state: %s, worker: %s, parent: #%s]" % (self.id, self.method, self.get_state_display(), self.worker, self.parent.id)
         return u"#%s [method: %s, state: %s, worker: %s]" % (self.id, self.method, self.get_state_display(), self.worker)
 
+    def __str__(self):
+        if self.parent:
+            return u"#%s [method: %s, state: %s, worker: %s, parent: #%s]" % (self.id, self.method, self.get_state_display(), self.worker, self.parent.id)
+        return u"#%s [method: %s, state: %s, worker: %s]" % (self.id, self.method, self.get_state_display(), self.worker)
```

Comment:
OK, this one is complex enough that you probably shouldn't duplicate the implementation with `__unicode__`.  It's not even that easy to verify by reading that the two are identical, since lines are quite long.

Could you please find a way to avoid duplicating this between `__str__` and `__unicode__`, e.g.  using six.python_2_unicode_compatible ?